cmd/sonobuoy/app/runner.go

```diff
@@ -0,0 +1,330 @@
+package app
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/util/exec"
+
+	"github.com/heptio/sonobuoy/pkg/client"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/heptio/sonobuoy/pkg/plugin/aggregation"
+)
+
+type runnerFlags struct {
+	genFlags
+	showAll         bool
+	skipPreflight   bool
+	pollingInterval time.Duration
+	statusLoopDelay time.Duration
+	retrievalDelay  time.Duration
+	timeout         time.Duration
+}
+
+var (
+	runnerflags    runnerFlags
+	sonobuoyClient *client.SonobuoyClient
+)
+
+func RunnerFlagSet(cfg *runnerFlags) *pflag.FlagSet {
+	runnerset := pflag.NewFlagSet("runner", pflag.ExitOnError)
+	// Default to detect since we need kubeconfig regardless
+	runnerset.AddFlagSet(GenFlagSet(&cfg.genFlags, DetectRBACMode, ConformanceImageVersionAuto))
+	AddSkipPreflightFlag(&cfg.skipPreflight, runnerset)
+
+	runnerset.DurationVar(
+		&cfg.timeout, "runner-timeout", 6*time.Hour,
+		"Length of time to give the runner before giving up. (Default: 6h)",
+	)
+	runnerset.DurationVar(
+		&cfg.pollingInterval, "polling-interval", 5*time.Minute,
+		"Duration of time between polling sonobuoy for status. (Default: 5m)",
+	)
+	runnerset.DurationVar(
+		&cfg.statusLoopDelay, "status-loop-delay", 1*time.Minute,
+		"Duration of time between starting tests and checking the status. (Default: 1m)",
+	)
+	runnerset.DurationVar(
+		&cfg.retrievalDelay, "retrieval-delay", 15*time.Second,
+		"Duration of time between getting completed status and retrieving results. (Default: 15s)",
+	)
+	runnerset.BoolVar(
+		&cfg.showAll, "show-all", false,
+		"Don't summarize plugin statuses, show all individually",
+	)
+
+	return runnerset
+}
+
+func (r *runnerFlags) Config() (*client.RunConfig, error) {
+	gencfg, err := r.genFlags.Config()
+	if err != nil {
+		return nil, err
+	}
+	return &client.RunConfig{
+		GenConfig: *gencfg,
+	}, nil
+}
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "runner",
+		Short: "Submits a sonobuoy run and manages the lifetime of that run",
```

Comment:
I'd like to see a longer description here. What does 'manage the lifetime'  mean? what are the expected outputs?